src/components/Tree/TreeItem.tsx

```diff
@@ -0,0 +1,141 @@
+import { defaultBehavior } from '../../lib/accessibility'
+import { Accessibility } from '../../lib/accessibility/types'
+
+import {
+  AutoControlledComponent,
+  childrenExist,
+  customPropTypes,
+  createShorthandFactory,
+} from '../../lib'
+import { ShorthandRenderFunction, ShorthandValue } from 'utils'
+import {
+  commonUIComponentPropTypes,
+  childrenComponentPropTypes,
+  contentComponentPropsTypes,
+} from '../../lib/commonPropTypes'
+import {
+  UIComponentProps,
+  ChildrenComponentProps,
+  ContentComponentProps,
+} from '../../lib/commonPropInterfaces'
+
+export interface TreeItemProps
+  extends UIComponentProps<any, any>,
+    ChildrenComponentProps,
+    ContentComponentProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * @default defaultBehavior
+   */
+  accessibility?: Accessibility
+
+  /** Initial activeIndex value. */
+  defaultOpen?: boolean
+
+  /** Array of props for sub tree. */
+  items?: ShorthandValue[]
+
+  /** Whether or not the subtree of the item is in the open state. */
+  open?: boolean
+
+  /**
+   * A custom render iterator for rendering each Accordion panel title.
+   * The default component, props, and children are available for each panel title.
+   *
+   * @param {React.ReactType} Component - The computed component for this slot.
+   * @param {object} props - The computed props for this slot.
+   * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+   */
+  renderItemTitle?: ShorthandRenderFunction
+
+  /** Properties for TreeTitle. */
+  title?: ShorthandValue
+}
+
+export interface TreeItemState {
+  open?: boolean
+}
+
+class TreeItem extends AutoControlledComponent<TreeItemProps, TreeItemState> {
+  static create: Function
+
+  static className = 'ui-tree__item'
+
+  static displayName = 'TreeItem'
+
+  static autoControlledProps = ['open']
+
+  static propTypes = {
+    ...commonUIComponentPropTypes,
+    ...childrenComponentPropTypes,
+    ...contentComponentPropsTypes,
+    accessibility: PropTypes.func,
+    defaultOpen: PropTypes.bool,
+    items: customPropTypes.collectionShorthand,
+    open: PropTypes.bool,
+    renderItemTitle: PropTypes.func,
+    title: customPropTypes.itemShorthand,
+  }
+
+  public static defaultProps = {
+    as: 'li',
+    accessibility: defaultBehavior,
+  }
+
+  handleTitleOverrides = predefinedProps => ({
+    onClick: (e, titleProps) => {
+      e.preventDefault()
+      this.trySetState({
+        open: !this.state.open,
+      })
+      _.invoke(predefinedProps, 'onClick', e, titleProps)
+    },
+  })
+
+  renderContent() {
+    const { items, title, renderItemTitle } = this.props
+    const { open } = this.state
+
+    return (
+      <>
+        {TreeTitle.create(title, {
+          defaultProps: {
+            open,
+            hasSubtree: !!(items && items.length),
+          },
+          render: renderItemTitle,
+          overrideProps: this.handleTitleOverrides,
+        })}
+        {items &&
```

Comment:
Should we still render a subtree if ```items``` is an empty array?
So maybe check ```items && items.length``` as you checked on the line 110. 
Event better to move this to a variable -> ```const hasSubtree = items && items.length```
and use it on
- line 110 - ```hasSubtree: hasSubtree```
- line 115 - ```{hasSubtree && open ....} ```